app/src/openxr/cpp/OpenXRPassthroughStrategy.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <memory>
+
+#include <openxr/openxr.h>
+#include "OpenXRLayers.h"
+
+namespace crow {
+
+class OpenXRPassthroughStrategy;
+typedef std::unique_ptr<OpenXRPassthroughStrategy> PassthroughStrategyPtr;
+
+class OpenXRPassthroughStrategy {
+public:
+enum class HandleEventResult { NoError, NonRecoverableError, NeedsReinit };
+virtual void initializePassthrough(XrSession) {}
+virtual bool usesCompositorLayer() const { return false; }
+virtual XrEnvironmentBlendMode environmentBlendModeForPassthrough() const = 0;
+virtual HandleEventResult handleEvent(const XrEventDataBaseHeader&) { return HandleEventResult::NoError; };
+virtual ~OpenXRPassthroughStrategy() = default;
+virtual bool isReady() const { return mIsInErrorState; };
+virtual OpenXRLayerPassthroughPtr createLayerIfSupported(VRLayerPassthroughPtr) const;
+protected:
+bool mIsInErrorState { false };
+};
+
+class OpenXRPassthroughStrategyNone : public OpenXRPassthroughStrategy {
+private:
+XrEnvironmentBlendMode environmentBlendModeForPassthrough() const override;
+};
+
+class OpenXRPassthroughStrategyFBExtension : public OpenXRPassthroughStrategy {
```

Comment:
Not a strong opinion, but perhaps we could use independent files for the different strategies. IF we do this, they are more explicit when using source code indexers (or just looking at the source files in our terminal).

Reply:
Sure, but the amount of code for some of them is basically 0, that's why I think as a first step it's more convenient to have them all together